### DISCUSSION TITLE
Why is the "-no-remote" option accepted by falkon but not lynx?

### DISCUSSION THREAD

**Discussion by Rosika2** [netblue30/firejail#4857] [Q&A] [answered]
Hi all,

... and a happy and healthy New Year.

I´ve got a question regarding the usage of the **"-no-remote"** parameter.

I experienced different behaviour when using this parameter with the browsers **lynx** and **falkon**.

I may run this command and it works well:

`firejail --disable-mnt --private --dns=1.1.1.1 --dns=9.9.9.9 falkon -no-remote`

Falkon opens up and can be used the normal way.

However: this command doesn´t work (just replacing **falkon** with **lynx**):

`firejail --disable-mnt --private --dns=1.1.1.1 --dns=9.9.9.9 lynx -no-remote`

Here the terminal gives me the following output:

```
lynx: Invalid Option: -no-remote
USAGE: lynx [options] [file]
Options are: [...]
```

So **"-no-remote"** seems to be interpreted as an option for **lynx** which is not available/valid.
But it seems to be valid with **falkon**.

Does anybody have an idea why that is?

Many thanks in advance and many greetings.

Rosika

P.S.:

Without the "-no-remote" option there´s no problem using lynx.

**Comment by kmk3** [comment, marked as answer]
@Rosika2 [on Jan 15](https://github.com/netblue30/firejail/discussions/4857#discussion-3802224):

> Hi all,
> 
> ... and a happy and healthy New Year.

Thanks, to you as well.

> I´ve got a question regarding the usage of the **"-no-remote"** parameter.
> 
> I experienced different behaviour when using this parameter with the browsers
> **lynx** and **falkon**.
> 
> I may run this command and it works well:
> 
> `firejail --disable-mnt --private --dns=1.1.1.1 --dns=9.9.9.9 falkon -no-remote`
> 
> Falkon opens up and can be used the normal way.
> 
> However: this command doesn´t work (just replacing **falkon** with **lynx**):
> 
> `firejail --disable-mnt --private --dns=1.1.1.1 --dns=9.9.9.9 lynx -no-remote`
> 
> Here the terminal gives me the following output:
> 
> ```
> lynx: Invalid Option: -no-remote
> USAGE: lynx [options] [file]
> Options are: [...]
> ```
> 
> So **"-no-remote"** seems to be interpreted as an option for **lynx** which
> is not available/valid. But it seems to be valid with **falkon**.
> 
> Does anybody have an idea why that is?

Every argument after the program name is passed to the program, so `-no-remote`
should not be interpreted by firejail in your examples.

Also, `-no-remote` is not an option of firejail.

I don't know about other browsers, but it is a valid option for Firefox.

From `firefox --help`:

```
Firefox options
  -h or --help       Print this message.
  -v or --version    Print Firefox version.
  --full-version     Print Firefox version, build and platform build ids.
  -P <profile>       Start with <profile>.
  --profile <path>   Start with profile at <path>.
  --migration        Start with migration wizard.
  --ProfileManager   Start with ProfileManager.
  --no-remote        Do not accept or send remote commands; implies
                     --new-instance.
  --new-instance     Open new instance, not a new window in running instance.
```

The only place where I found `--no-remote` mentioned on firejail is on
firejail(1):

```
       --name=name
	      Set sandbox name. Several options, such as --join and --shutdown,
	      can use this name to identify a sandbox.

	      In  case the name supplied by the user is already in use by
	      another sandbox, Firejail will assign a new name as "name-PID",
	      where PID is the process ID of the sandbox. This functionality
	      can be disabled at run time in /etc/firejail/firejail.config
	      file, by setting "name-change" flag to "no".

              Example:
              $ firejail --name=browser firefox &
              $ firejail --name=browser --private firefox --no-remote &
              $ firejail --list
              1198:netblue:browser:firejail --name=browser firefox
              1312:netblue:browser-1312:firejail --name=browser --private firefox --no-remote
```

In the above example, it is used in order to be able to open multiple Firefox
instances under the same user (since by default new links are opened in the
same instance).

AFAIK lynx does not use a user-wide instance by default like Firefox, so there
is no need for it to support a `--no-remote` option.

**Reply by rusty-snake** [reply]
FWIW: Falkon supports `--no-remote` too: https://github.com/KDE/falkon/commit/53d4f00c3697f19bc7a86d4b349f8984561b0b4c.

**Comment by Rosika2** [comment]
Hi all and thanks a lot for your answers,

@kmk3:

Thank you for pointing out the matter in such a detailed way.

I was under the misapprehension that the "-no-remote" option was somehow connected to the firejail command.
Which in effect cannot be as you pointed out:

> Every argument after the program name is passed to the program, so -no-remote
> should not be interpreted by firejail in your examples.

I really should´ve known better...

I looked it up in the man-pages of **firefox** and indeed I also found it there:

> -no-remote
>               Don't  connect to any other running instances of firefox. Use this if you want
>               to run firefox in an entirely new process. By default, firefox will delegate a
>               command to an already running instance.

So everything´s clear now.

From all that i deduce I don´t have to use the "-no-remote" option at all for my scenario:

- Firejail is already running in a sandbox
- At the same time I want to run either falkon or lynx in sandbox
- As (firejial´s) "no-remote" option refers to running instances if the same process (i.e. **firefox**) it won´t have any implications on falkon/lynx (which are different processes altogether).

Be thanked once more for your kind help.

@rusty-snake:

Hi and thanks a lot to you as well,

tnx also for the link which is quite helpful.

The man-pages of falkon (I looked them up on my system) only provide a short description of **falkon** and don´t list any options/parameters.

Good to know **falkon** supports the "no-remote" option as well.

Many thanks to both of you for your kind help ...
... and many greetings from Rosika
